Update how-to-authenticate-an-imap-pop-smtp-application-by-using-oaut… (#283)
MicrosoftDocs/office-developer-exchange-docs: docs/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth.md

### IMAP Protocol Exchange

> To authenticate an IMAP server connection, the client will have to respond with an `AUTHENTICATE` command in the following format:

Reviewer: ```suggestion
To authenticate an IMAP server connection, the client must respond with an `AUTHENTICATE` command in the following format:
```

S: 535 5.7.3 Authentication unsuccessful [SN2PR00CA0018.namprd00.prod.outlook.com]
```

> ## Using client credentials grant flow to authenticate IMAP and POP connections

Reviewer: ```suggestion
## Use client credentials grant flow to authenticate IMAP and POP connections
```

> Exchange service principals are used to enable applications to access Exchange mailboxes with client credentials flow with the POP and IMAP protocols.

Reviewer: ```suggestion
Use Exchange service principals to enable applications to access Exchange mailboxes with client credentials flow with the POP and IMAP protocols.
```

### Get tenant admin consent

For each tenant you would like to access Exchange mailboxes via POP or IMAP, your AAD application must get tenant admin consent. Learn more about [tenant admin consent process](/azure/active-directory/develop/v2-permissions-and-consent) here.

In your OAuth 2.0 tenant authorization request, the `scope` query parameter should be `https://ps.outlook.com/.default` for both the POP and IMAP application scopes.

Your OAuth 2.0 authorization request URL should look something like this:

```
https://login.microsoftonline.com/{tenant}/v2.0/adminconsent?client_id=CLIENT_ID_HERE&redirect_uri=REDIRECT_URI_HERE&scope=https://ps.outlook.com/.default
```

### Registering service principals in Exchange

Once your AAD application is consented by a tenant admin, the tenant admin must register your AAD application's service principal in Exchange via Exchange Online PowerShell. This is enabled by the `New-ServicePrincipal` cmdlet. (TODO: Link to New-ServicePrincipal cmdlet reference when released)

Here's an example of registering an AAD application's service principal in Exchange:

```

> New-ServicePrincipal –Identity <Service principal object ID in AAD> -AppId <Application ID in AAD> [-Organization <Org ID>]

Reviewer: ```suggestion
New-ServicePrincipal –Identity <Service principal object ID in Azure AD> -AppId <Application ID in Azure AD> [-Organization <Org ID>]
```

```

Your tenant admin can now add the specific mailboxes in the tenant that will be allowed to be access by your application. This is done with the [`Add-MailboxPermission` cmdlet](/powershell/module/exchange/add-mailboxpermission).

> Here's an example of giving your application access to one mailbox:

Reviewer: ```suggestion
The following an example of giving your application access to one mailbox:
```
